[![Awesome](https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)](https://github.com/sindresorhus/awesome)
[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](http://makeapullrequest.com) ![GitHub stars](https://img.shields.io/github/stars/XuYuanchi/Awesome-Reproducible-Analysis-of-Single-Cell-Papers?color=yellow&label=Stars) ![GitHub forks](https://img.shields.io/github/forks/XuYuanchi/Awesome-Reproducible-Analysis-of-Single-Cell-Papers?color=blue&label=Forks) 
# Awesome-Reproducible-Analysis-of-Single-Cell-Papers
A curated list of awesome reproducible papers in single-cell omics. 

The listed papers provide the code necessary for reproducing the figures or results.

## Table of Contents

- [Guidelines for Single-Cell Analysis](#guidelines-for-single-cell-analysis)
- [scRNA-seq](#scrna-seq)
- [scATAC-seq](#scatac-seq)
- [scHi-C](#schi-c)
- [Spatial Transcriptome](#spatial-transcriptome)
- [Multiomics](#multiomics)

## Guidelines for Single-Cell Analysis
1. [2023 Nature Reviews Genetics]**Best practices for single-cell analysis across modalities**[[paper]](https://www.nature.com/articles/s41576-023-00586-w)[[code]](https://www.nature.com/articles/s41576-023-00586-w)
## scRNA-seq
1. [Nature]**Universal recording of immune cell interactions in vivo**[[paper]](https://www.nature.com/articles/s41586-024-07134-4)[[code]](https://github.com/pritykinlab/ulipstic-analysis/tree/main)
2. [2023 Nature Medicine]**An integrated cell atlas of the lung in health and disease**[[paper]](https://www.nature.com/articles/s41591-023-02327-2#Abs1)[[code]](https://github.com/LungCellAtlas/HLCA_reproducibility)
3. [2023 Cancer Cell]**Tumor monocyte content predicts immunochemotherapy outcomes in esophageal adenocarcinoma**[[paper]](https://www.sciencedirect.com/science/article/pii/S1535610823002167?via%3Dihub)[[code]](https://zenodo.org/record/8083316)
4. [2022 Molecular Neurodegeneration]**Guidelines for bioinformatics of single-cell sequencing data analysis in Alzheimer’s disease: review, recommendation, implementation and application**[[paper]](https://molecularneurodegeneration.biomedcentral.com/articles/10.1186/s13024-022-00517-z#Abs1)[[code]](https://github.com/songw01/AD_scRNAseq_companion)
5. [2022 Nature Cancer]**Breast cancer prevention by short-term inhibition of TGFβ signaling**[[paper]](https://www.nature.com/articles/s41467-022-35043-5)[[code]](https://github.com/csimona/tumor-prevention-rat-scRNAseq)
6. [2022 Nature Cell Biology]**Single-cell analysis of endometriosis reveals a coordinated transcriptional programme driving immunotolerance and angiogenesis across eutopic and ectopic tissues**[[paper]](https://www.nature.com/articles/s41556-022-00961-5)[[code]](https://github.com/TheJacksonLaboratory/endometriosis-scrnaseq/tree/main)
7. [2022 Nature Neuroscience]**Dissection of artifactual and confounding glial signatures by single-cell sequencing of mouse and human brain**[[paper]](https://www.nature.com/articles/s41593-022-01022-8)[[code]](https://github.com/samuel-marsh/Marsh_et-al_2022_scRNAseq_Dissociation_Artifacts)
8. [2021 Nature]**Global absence and targeting of protective immune states in severe COVID-19**[[paper]](https://www.nature.com/articles/s41586-021-03234-7)[[code]](https://github.com/UCSF-DSCOLAB/combes_et_al_COVID_2020?tab=readme-ov-file)
9. [2021 Nature Communications]**Single-cell transcriptomics captures features of human midbrain development and dopamine neuron diversity in brain organoids**[[paper]](https://www.nature.com/articles/s41467-021-27464-5)[[code]](https://github.com/ParmarLab/scRNA-seq_silk_organoids/tree/main)
## scATAC-seq
1. [2023 Nature]**Molecular features driving cellular complexity of human brain evolution**[[paper]](https://www.nature.com/articles/s41586-023-06338-4#Abs1)[[code]](https://github.com/konopkalab/Comparative_snATAC_snRNA)
2. [2023 Nature]**Single-cell analysis of chromatin accessibility in the adult mouse brain**[[paper]](https://www.nature.com/articles/s41586-023-06824-9)[[code]](https://github.com/beyondpie/CEMBA_wmb_snATAC)
3. [2023 Narure]**Epigenetic regulation during cancer transitions across 11 tumour types**[[paper]](https://www.nature.com/articles/s41586-023-06682-5)[[code]](https://github.com/ding-lab/PanCan_snATAC_publication/tree/main)
4. [2023 Nature]**Organization of the human intestine at single-cell resolution**[[paper]](https://www.nature.com/articles/s41586-023-05915-x#Abs1)[[code]](https://github.com/winstonbecker/scColonHuBMAP/)
5. [2023 biorxiv]**Transcription factor stoichiometry, motif affinity and syntax regulate single-cell chromatin dynamics during fibroblast reprogramming to pluripotency**[[paper]](https://www.biorxiv.org/content/10.1101/2023.10.04.560808v1)[[code]](https://github.com/kundajelab/scATAC-reprog)
6. [2023 Nature Biotechnology]**SEACells infers transcriptional and epigenomic cellular states from single-cell genomics data**[[paper]](https://www.nature.com/articles/s41587-023-01716-9#Abs1)[[code]](https://github.com/dpeerlab/SEACellsReproducibility/tree/main/notebooks)
7. [2023 eLife]**The single-cell chromatin accessibility landscape in mouse perinatal testis development**[[paper]](https://elifesciences.org/articles/75624#abstract)[[code]](https://github.com/liaojinyue/mouse_testis_scATAC)
8. [2022 Nature Biotechnology]**Variant to function mapping at single-cell resolution through network propagation**[[paper]](https://www.nature.com/articles/s41587-022-01341-y#Abs1)[[code]](https://github.com/sankaranlab/SCAVENGE-reproducibility)
9. [2022 Nature Communications]**Single-cell chromatin profiling of the primitive gut tube reveals regulatory dynamics underlying lineage fate decisions**[[paper]](https://www.nature.com/articles/s41467-022-30624-w#Abs1)[[code]](https://github.com/zang-lab/Single-cell-chromatin-profiling-of-the-primitive-gut-tube/tree/main)
10. [2021 Science]**Developmental and evolutionary dynamics of cis-regulatory elements in mouse cerebellar cells**[[paper]](https://www.science.org/doi/10.1126/science.abg4696)[[code]](https://github.com/ioansarr/mouse-cerebellum-atac/tree/master)
11. [2021 Nature Biotechnology]**Integrative single-cell analysis of allele-specific copy number alterations and chromatin accessibility in cancer**[[paper]](https://www.nature.com/articles/s41587-021-00911-w#Abs1)[[code]](https://doi.org/10.24433/CO.2295856.v1)
12. [2021 Nature Communications]**High-throughput single-cell chromatin accessibility CRISPR screens enable unbiased identification of regulatory networks in cancer**[[paper]](https://www.nature.com/articles/s41467-021-23213-w#Sec1)[[code]](https://github.com/GreenleafLab/SpearATAC_MS_2021/tree/main)

## scHi-C
1. **Notes on single-cell Hi-C technologies, tools, and data**[[code]](https://github.com/mdozmorov/scHiC_notes)
2. [2023 Nature Communications]**Single cell Hi-C identifies plastic chromosome conformations underlying the gastrulation enhancer landscape**[[paper]](https://www.nature.com/articles/s41467-023-39549-4)[[code]](https://github.com/tanaylab/scHiC_embryo)

## Spatial Transcriptome
1. [2023 Science Advances]**Single-cell and spatial transcriptome analysis reveals the cellular heterogeneity of liver metastatic colorectal cancer**[[paper]](https://www.science.org/doi/full/10.1126/sciadv.adf5464)[[code]](https://github.com/jalon9358/LianLab_CRCLM)
2. [2023 Nature Genetics]**CellCharter reveals spatial cell niches associated with tissue remodeling and cell plasticity**[[paper]](https://www.nature.com/articles/s41588-023-01588-4)[[code]](https://github.com/CSOgroup/cellcharter_analyses?tab=readme-ov-file)
3. [2023 Cancer Research]**Spatial Transcriptomics Depict Ligand–Receptor Cross-talk Heterogeneity at the Tumor-Stroma Interface in Long-Term Ovarian Cancer Survivors**[[paper]](https://aacrjournals.org/cancerres/article-abstract/83/9/1503/726052/Spatial-Transcriptomics-Depict-Ligand-Receptor?redirectedFrom=fulltext)[[code]](https://codeocean.com/capsule/1912679/tree/v1)
4. [2022 Nature]**Spatial genomics maps the structure, nature and evolution of cancer clones**[[paper]](https://www.nature.com/articles/s41586-022-05425-2)[[code]](https://github.com/gerstung-lab/BaSISS/tree/main)
5. [2022 Nature Methods]**Squidpy: a scalable framework for spatial omics analysis**[[paper]](https://www.nature.com/articles/s41592-021-01358-2)[[code]](https://github.com/theislab/squidpy_reproducibility/tree/master)
6. [2022 bioRxiv]**Harmonized single-cell landscape, intercellular crosstalk and tumor architecture of glioblastoma**[[paper]](https://www.biorxiv.org/content/10.1101/2022.08.27.505439v1)[[code]](https://github.com/ccruizm/GBmap/tree/main)
7. [2021 Nature Neuroscience]**Transcriptome-scale spatial gene expression in the human dorsolateral prefrontal cortex**[[paper]](https://www.nature.com/articles/s41593-020-00787-0)[[code]](https://github.com/LieberInstitute/HumanPilot/tree/master)

## Multiomics
1. [2024 Nature Genetics]**Multi-omic profiling of clear cell renal cell carcinoma identifies metabolic reprogramming associated with disease progression**[[paper]](https://www.nature.com/articles/s41588-024-01662-5)[[code]](https://github.com/AndersonHu85/ccRCC_multiomics)
2. [2023 Nature Biotechnology]**Multimodal spatiotemporal phenotyping of human retinal organoid development**[[paper]](https://www.nature.com/articles/s41587-023-01747-2#Abs1)[[code]](https://github.com/quadbio/spatial_multimodal_retinal_organoid/tree/main)
3. [2023 Nature Genetics]**Integrated single-cell chromatin and transcriptomic analyses of human scalp identify gene-regulatory programs and critical cell types for hair and skin diseases**[[paper]](https://www.nature.com/articles/s41588-023-01445-4)[[code]](https://github.com/GreenleafLab/scScalpChromatin)
4. [2023 Nature Communications]**Deciphering early human pancreas development at the single-cell level**[[paper]](https://www.nature.com/articles/s41467-023-40893-8#Abs1)[[code]](https://github.com/zhuoma888/fetal_pancreas)
5. [2023 Nature Communications]**Single-cell transcriptomics and epigenomics unravel the role of monocytes in neuroblastoma bone marrow metastasis**[[paper]](https://www.nature.com/articles/s41467-023-39210-0#Abs1)[[code]](https://github.com/csbg/neuroblastoma)
6. [2022 Nature Genetics]**Single-cell analyses define a continuum of cell state and composition changes in the malignant transformation of polyps to colorectal cancer**[[paper]](https://www.nature.com/articles/s41588-022-01088-x#Abs1)[[code]](https://github.com/winstonbecker/scCRC_continuum/tree/main)
7. [2022 Nature Genetics]**K27M in canonical and noncanonical H3 variants occurs in distinct oligodendroglial cell lineages in brain midline gliomas**[[paper]](https://www.nature.com/articles/s41588-022-01205-w#Abs1)[[code]](https://github.com/fungenomics/HGG-oncohistones)
8. [2022 Nature neuroscience]**Multimodal profiling of the transcriptional regulatory landscape of the developing mouse cortex identifies Neurog2 as a key epigenome remodeler**[[paper]](https://www.nature.com/articles/s41593-021-01002-4#Abs1)[[code]](https://github.com/BonevLab/Noack_et_al_NatNeuro2021/)
9. [2022 Cell Discovery]**Single-cell multiomics analysis reveals regulatory programs in clear cell renal cell carcinoma**[[paper]](https://www.nature.com/articles/s41421-022-00415-0#Abs1)[[code]](https://github.com/Dragonlongzhilin/RenalTumor/tree/main)
10. [2021 Molecular Cell]**A multi-omic single-cell landscape of human gynecologic malignancies**[[paper]](https://www.sciencedirect.com/science/article/pii/S109727652100842X?via%3Dihub)[[code]](https://github.com/RegnerM2015/scENDO_scOVAR_2020/tree/main)
11. [2019 Nature biotechnology]**Single-cell multiomic analysis identifies regulatory programs in mixed-phenotype acute leukemia**[[paper]](https://www.nature.com/articles/s41587-019-0332-7#Abs1)[[code]](https://github.com/GreenleafLab/MPAL-Single-Cell-2019/tree/master)
